src/components/grid-selector.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+function GridSelector({ items, selectedItemKey, onSelectionChange }) {
+  const listRef = useRef();
+
+  useEffect(() => {
+    const selectedElement = listRef.current?.querySelector(`[data-item-key="${selectedItemKey.replaceAll('"', '\\"')}"]`);
+    if (selectedElement && selectedElement !== document.activeElement) {
+      selectedElement.focus();
+    }
+  }, [selectedItemKey]);
+
+  const handleFocusCapture = event => {
+    const itemKey = event.target.getAttribute('data-item-key');
+    if (itemKey && itemKey !== selectedItemKey) {
+      onSelectionChange?.(itemKey, false);
+    }
+  };
+
+  return (
+    <ul ref={listRef} className="GridSelector" onFocusCapture={handleFocusCapture}>
+      {items.map((item, index) => (
+        <li
+          key={item.key}
+          tabIndex={index + 1}
+          data-item-key={item.key}
+          className={classNames('GridSelector-item', { 'GridSelector-item--selected': item.key === selectedItemKey })}
```

Comment:
This class matches better as a component state, `is-selected`, I feel.

Reply:
Correct, changed it.